aspnetcore/blazor/javascript-interoperability/import-export-interop.md

```diff
@@ -0,0 +1,401 @@
+```
+
+Load the module in `Program.cs` before <xref:Microsoft.AspNetCore.Components.WebAssembly.Hosting.WebAssemblyHost.RunAsync%2A?displayProperty=nameWithType> is called:
+
+```csharp
+await JSHost.ImportAsync("Interop", "../js/interop.js");
+```
+
+`Pages/CallJavaScript2.razor`:
+
+```razor
+@page "/call-javascript-2"
+@using BlazorSample.JavaScriptInterop
+
+<h1>
+    .NET JS <code>[JSImport]</code>/<code>[JSExport]</code> Interop 
+    (Call JS Example 2)
+</h1>
+
+@(message is not null ? message : string.Empty)
+
+@code {
+    private string? message;
+
+    protected override void OnInitializedAsync()
+    {
+        message = Interop.GetWelcomeMessage();
+    }
+}
+```
+
+`Pages/CallDotNet2.razor`:
+
+```razor
+@page "/call-dotnet-2"
+@using BlazorSample.JavaScriptInterop
+
+<h1>
+    .NET JS <code>[JSImport]</code>/<code>[JSExport]</code> Interop  
+    (Call .NET Example 2)
+</h1>
+
+<p>
+    <span id="result">.NET method not executed</span>
+</p>
+
+@code {
+    protected override void OnAfterRender(bool firstRender)
+    {
+        Interop.SetWelcomeMessage();
+    }
+}
+```
+
+> [!IMPORTANT]
+> In this section's example, JS interop is used to mutate a DOM element *purely for demonstration purposes* after the component is rendered in [`OnAfterRender`](xref:blazor/components/lifecycle#after-component-render-onafterrenderasync). Typically, you should only mutate the DOM with JS when the object doesn't interact with Blazor. The approach shown in this section is similar to cases where a third-party JS library is used in a Razor component, where the component interacts with the JS library via JS interop, the third-party JS library interacts with part of the DOM, and Blazor isn't involved directly with the DOM updates to that part of the DOM. For more information, see <xref:blazor/js-interop/index#interaction-with-the-document-object-model-dom>.
+
+## Additional example
+
+For an additional example of the JS interop techniques described in this article, see the *Blazor WASM demo of new JSInterop* sample app:
+
+* [Reference source (`pavelsavara/blazor-wasm-hands-pose` GitHub repository)](https://github.com/pavelsavara/blazor-wasm-hands-pose)
+* [Live demonstration](https://pavelsavara.github.io/blazor-wasm-hands-pose/)
+
+> [!NOTE]
+> The [`pavelsavara/blazor-wasm-hands-pose` GitHub repository](https://github.com/pavelsavara/blazor-wasm-hands-pose) isn't owned, maintained, or supported by the .NET foundation or Microsoft.
+>
+> The *Blazor WASM demo of new JSInterop* sample app uses a public JS library from [MediaPipe](https://www.mediapipe.dev/), which isn't owned, maintained, or supported by the .NET foundation or Microsoft.
+
+## Additional resources
+
+<!--
+
+    HOLD: WAIT FOR CLIENT-SIDE ARTICLE TO MERGE
+    
+    * <xref:client-side/import-export-interop>
+
+-->
+
+* [Use .NET from any JavaScript app in .NET 7](https://devblogs.microsoft.com/dotnet/use-net-7-from-any-javascript-app-in-net-7/)
```

Comment:
I prefer to avoid links to blog post content as the content is point-in-time and can quickly go stale.

```suggestion
```